assets/js/base/components/price-slider/index.js

```diff
@@ -26,321 +36,269 @@ const PriceSlider = ( {
-					ref={ maxRange }
-					value={ value.max || 0 }
-					{ ...rangeInputProps }
-				/>
-			</Fragment>
-		);
-	};
+	const priceInputOnChange = useCallback(
+		( event ) => {
+			const newValue = event.target.value.replace( /[^0-9.-]+/g, '' );
+			const isMin = event.target.classList.contains(
+				'wc-block-price-filter__amount--min'
+			);
+			if ( isMin ) {
+				setFormattedMinPrice(
+					formatCurrencyForInput(
+						newValue,
+						priceFormat,
+						currencySymbol
+					)
+				);
+			} else {
+				setFormattedMaxPrice(
+					formatCurrencyForInput(
+						newValue,
+						priceFormat,
+						currencySymbol
+					)
+				);
+			}
+		},
+		[ priceFormat, currencySymbol ]
+	);
 
-	const minInput = useRef();
-	const maxInput = useRef();
-	const minRange = useRef();
-	const maxRange = useRef();
 	const classes = classnames(
 		'wc-block-price-filter',
 		showInputFields && 'wc-block-price-filter--has-input-fields',
 		showFilterButton && 'wc-block-price-filter--has-filter-button',
 		isLoading && 'is-loading'
 	);
+
 	return (
 		<div className={ classes }>
 			<div
 				className="wc-block-price-filter__range-input-wrapper"
 				onMouseMove={ findClosestRange }
 				onFocus={ findClosestRange }
 			>
-				{ ! isLoading && renderRangeInputs() }
+				{ ! isLoading && (
+					<Fragment>
+						<div
+							className="wc-block-price-filter__range-input-progress"
+							style={ getProgressStyle }
+						/>
+						<input
+							type="range"
+							className="wc-block-price-filter__range-input wc-block-price-filter__range-input--min"
+							aria-label={ __(
+								'Filter products by minimum price',
+								'woo-gutenberg-products-block'
+							) }
+							value={ minPrice || 0 }
+							onChange={ rangeInputOnChange }
+							step={ step }
+							min={ minConstraint }
+							max={ maxConstraint }
+							ref={ minRange }
+						/>
+						<input
+							type="range"
+							className="wc-block-price-filter__range-input wc-block-price-filter__range-input--max"
+							aria-label={ __(
+								'Filter products by maximum price',
+								'woo-gutenberg-products-block'
+							) }
+							value={ maxPrice || 0 }
+							onChange={ rangeInputOnChange }
+							step={ step }
+							min={ minConstraint }
+							max={ maxConstraint }
+							ref={ maxRange }
+						/>
+					</Fragment>
+				) }
 			</div>
 			<div className="wc-block-price-filter__controls">
-				{ showInputFields ? renderTextInputs() : renderText() }
-				{ showFilterButton && renderSubmit() }
+				{ showInputFields ? (
+					<PriceInput
+						disabled={ isLoading }
+						onChange={ priceInputOnChange }
+						onBlur={ priceInputOnBlur }
+						minPrice={ formattedMinPrice }
+						maxPrice={ formattedMaxPrice }
+					/>
+				) : (
+					<PriceLabel
+						minPrice={ formattedMinPrice }
+						maxPrice={ formattedMaxPrice }
+					/>
+				) }
+				{ showFilterButton && (
+					<SubmitButton
+						disabled={ isLoading }
+						onClick={ triggerChange }
+					/>
+				) }
```

Comment:
Love this change with the extracted components!  Thanks 👍 .  I think it is much more cleaner and readable now.